Tasks/JavaToolInstallerV0/chocoJavaInstall.ps1

```diff
@@ -0,0 +1,89 @@
+param (
+	[string]$VersionSpec
+    )
+
+# Set new machine path 
+function Set-MachinePath{
+    [CmdletBinding()]
+    param(
+        [string]$NewPath
+    )
+    Set-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name Path -Value $NewPath
+    return $NewPath
+}
+
+# Get machine path from registry
+function Get-MachinePath{
+    [CmdletBinding()]
+    param(
+
+    )
+    $currentPath = (Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name PATH).Path
+    return $currentPath
+}
+
+$jdkVersion = ("jdk" + $VersionSpec)
+$jdkMatchString = ('jdk*' + $VersionSpec + '*')
+$extendedJavaHome = ('JAVA_HOME_' + $VersionSpec + '_X64')
+
+Write-Host "Installing chocolatey"
+$chocoExePath = 'C:\ProgramData\Chocolatey\bin'
+
+if ($($env:Path).ToLower().Contains($($chocoExePath).ToLower())) {
+    Write-Host "Chocolatey found in PATH, skipping install..."
+}
+else {
+	# Add to system PATH
+	$systemPath = [Environment]::GetEnvironmentVariable('Path', [System.EnvironmentVariableTarget]::Machine)
+	$systemPath += ';' + $chocoExePath
+	[Environment]::SetEnvironmentVariable("PATH", $systemPath, [System.EnvironmentVariableTarget]::Machine)
```

Comment:
Does chocolatey.org/install.ps1 not update the machine PATH? If you have to do this manually, I would do it *after* you install chocolately. Currently if this task failed on line 51, it's not re-runnable. On the second execution it will find  'C:\ProgramData\Chocolatey\bin' in the PATH and not bother to install it.